### ISSUE TITLE
Quotes could be added to error messages to improve clarity

### ISSUE THREAD

**Issue by IanVS** [eslint/eslint#4313]
There are a few rules which could stand to use some quotes to mark the token being flagged (Similar to https://github.com/eslint/eslint/issues/3526).  These are:

### no-shadow

```js
/*eslint no-shadow: 2*/
var x = 1; 
function a(x) {
    return ++x; 
}
```

Gives:
```
4:12  error  x is already declared in the upper scope  no-shadow
```

Would be better as:
```
4:12  error  'x' is already declared in the upper scope  no-shadow
```

### no-use-before-define

```js
/*eslint no-use-before-define: 2*/

a++; 
var a=19;
```

Gives:
```
3:1  error  a was used before it was defined  no-use-before-define
```

Would be better as:
```
3:1  error  'a' was used before it was defined  no-use-before-define
```

### no-warning-comments

```js
/*eslint no-warning-comments: [2, { "terms": ["eslint"] }]*/

/* eslint one-var: 2 */
```

Gives:
```
3:1  error  Unexpected eslint comment  no-warning-comments
```

Would be better as:
```
3:1  error  Unexpected 'eslint' comment  no-warning-comments
```

### spaced-comment

```js
/*eslint spaced-comment: 2*/
```

Gives:
```
1:1  error  Expected space or tab after /* in comment  spaced-comment
```

Would be better as:
```
1:1  error  Expected space or tab after '/*' in comment  spaced-comment
```

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
:+1:
